test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,521 @@
+
+        if let Some(priority_value) = value.get("priority").and_then(|v| v.as_str()) {
+            uattributes.priority = UPriority::from_str(priority_value)
+                .unwrap_or_else(|| {
+                    // Handle the case where the conversion fails
+                    error!("Error:: Something wrong with priority field");
+                    UPriority::UPRIORITY_UNSPECIFIED
+                })
+                .into();
+        } else {
+            error!("Error:pririty value is not string!");
+        }
+
+        dbg!("uattributes.priority: {:?}", uattributes.priority);
+
+        if let Some(type_value) = value.get("type").and_then(|v| v.as_str()) {
+            uattributes.type_ = UMessageType::from_str(type_value)
+                .unwrap_or_else(|| {
+                    // Handle the case where the conversion fails
+                    error!("Error: Something wrong with type field");
+                    UMessageType::UMESSAGE_TYPE_UNSPECIFIED
+                })
+                .into();
+        } else {
+            error!("Error: type value is not string!");
+        }
+
+        dbg!("uattributes.type_: {:?}", uattributes.type_);
+
+        if let Some(source_value) = value.get("source") {
+            if let Ok(wrapper_uri) = serde_json::from_value::<WrapperUUri>(source_value.clone()) {
+                uattributes.source = MessageField(Some(Box::new(wrapper_uri.0)));
+            }
+        };
+        if let Some(sink_value) = value.get("sink") {
+            if let Ok(wrapper_uri) = serde_json::from_value::<WrapperUUri>(sink_value.clone()) {
+                uattributes.sink = MessageField(Some(Box::new(wrapper_uri.0)));
+            }
+        };
+
+        let Ok(id) = parse_uuid(&value, "id") else {
+            let err_msg = "Error parsing UUID id: ".to_string();
+            return Err(serde::de::Error::custom(err_msg));
+        };
+
+        uattributes.id = MessageField(Some(Box::new(id)));
+
+        if let Some(ttl) = value.get("ttl").and_then(|ttl| ttl.as_str()) {
+            if let Ok(parsed_ttl) = ttl.parse::<u32>() {
+                uattributes.ttl = parsed_ttl.into();
+            } else {
+                error!("Error: Failed to parse _ttl as u32");
+            }
+        };
+
+        if let Some(permission_level) = value
+            .get("permission_level")
+            .and_then(|permission_level| permission_level.as_str())
+        {
+            if let Ok(parsed_permission_level) = permission_level.parse::<u32>() {
+                uattributes.permission_level = Some(parsed_permission_level);
+            } else {
+                error!("Error: Failed to parse permission_level as u32");
+            }
+        };
+
+        if let Some(commstatus_value) = value
+            .get("commstatus")
+            .and_then(|commstatus_value| commstatus_value.as_str())
+        {
+            uattributes.commstatus = Some(UCode::from_str(commstatus_value).unwrap().into());
```

Comment:
would seek help removing this unwrap

Reply:
Here's one way of going about it:
```rust
if let Some(commstatus_str) = value.get("commstatus").and_then(|v| v.as_str()) {
  match UCode::from_str(commstatus_str) {
    Some(code) => uattributes.commstatus = Some(code.into()),
    None => error!("Failed to parse commstatus string into UCode."),
  }
} else {
  error!("Failed to extract commstatus as a string.");
}
```
I like this a little better than continuing to chain further.

We use an `if let` and then a `match` to handle the error conditions.

Reply:
Makes sense, going to try it out